### PR TITLE
Added public API documentation for log4cxx and spdlog

### PR DESCRIPTION
**Pull request by ahcorde**
Added public API documentation for log4cxx and spdlog.

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h [hunk 1]
```diff
@@ -23,17 +23,54 @@ extern "C" {
...
 RCL_LOGGING_PUBLIC
 rcl_logging_ret_t rcl_logging_external_initialize(
   const char * config_file,
   rcutils_allocator_t allocator);
 
+/// Free the resources allocated for the log4cxx external logging system.
+/**
+ * This puts the system into a state equivalent to being uninitialized.
+ *
+ * \return always RCL_LOGGING_RET_OK.
```
Comment (brawner) [inline rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h:47]: Is this always the intent of the API, or does it just return OK currently in the implementation?

File: rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h [hunk 2]
```diff
@@ -23,17 +23,54 @@ extern "C" {
...
 
+/// Set the severity level for a logger.
+/**
+ * This function sets the severity level for the specified logger.
+ * If the name provided is an empty string or NULL it will change the level of
+ * the root logger.
+ *
+ * \param[in] name The name of the logger.
+ *   Must be a NULL terminated c string or NULL.
+ * \param[in] level The severity level to be used for the specified logger.
```
Comment (brawner) [inline rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h:71]: Where are these severity levels defined? Maybe you can link to them or tag them?

File: rcl_logging_spdlog/include/rcl_logging_spdlog/logging_interface.h [hunk 3]
```diff
@@ -23,17 +23,51 @@ extern "C" {
...
 RCL_LOGGING_PUBLIC
 rcl_logging_ret_t rcl_logging_external_initialize(
   const char * config_file,
   rcutils_allocator_t allocator);
 
+/// Free the resources allocated for the spdlog external logging system.
+/**
+ * This puts the system into a state equivalent to being uninitialized.
+ *
+ * \return always RCL_LOGGING_RET_OK.
```
Comment (brawner) [inline rcl_logging_spdlog/include/rcl_logging_spdlog/logging_interface.h:47]: Same question

File: rcl_logging_spdlog/include/rcl_logging_spdlog/logging_interface.h [hunk 4]
```diff
@@ -23,17 +23,51 @@ extern "C" {
...
 void rcl_logging_external_log(int severity, const char * name, const char * msg);
 
+/// Set the severity level for a logger.
+/**
+ * This function sets the severity level for the specified logger.
+ * If the name provided is an empty string or NULL it will change the level of
+ * the root logger.
+ *
+ * \param[in] name The name of the logger (unused).
+ * \param[in] level The severity level to be used for the specified logger.
```
Comment (brawner) [inline rcl_logging_spdlog/include/rcl_logging_spdlog/logging_interface.h:68]: Same question as other file

File: rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h (hunk 1 above)
Comment (ahcorde) [reply rcl_logging_log4cxx/include/rcl_logging_log4cxx/logging_interface.h:47]: the current implementation returns OK always.

File: rcl_logging_spdlog/include/rcl_logging_spdlog/logging_interface.h (hunk 3 above)
Comment (ahcorde) [reply rcl_logging_spdlog/include/rcl_logging_spdlog/logging_interface.h:47]: the current implementation returns always OK

**Pull request merged**